src/main/kotlin/compiler/Parser.kt

```diff
@@ -74,6 +74,7 @@ class Parser(inputTokens: List<Token>) {
         pWhileLoop()?.also { return it }
         pReturn()?.also { return it }
         pFail()?.also { return it }
+        pWhen(asStatement = true)?.also { return node(N_EXPRSTATEMENT(it)) }
```

Comment:
We need to parse 'when' differently in a statement context, because that version doesn't require an else, and also needs to clean up the stack (since it won't be consumed as a value).